server/app/controllers/TaskSolver.scala

```diff
@@ -26,20 +29,28 @@ class TaskSolver @Inject()(executor: RuntimeSuiteExecutor, dao: Dao, val message
 
   val solutionForm = Form {
     mapping(
-      solution -> nonEmptyAndDiffer(from = solutionTemplateText)
+      solution -> nonEmptyText
     )(SolutionForm.apply)(SolutionForm.unapply)
   }
 
-  def getTask = Action { implicit request =>
-    Ok(views.html.task(taskDescriptionText, solutionForm.fill(SolutionForm(solutionTemplateText))))
+  def getTask(year: Long, taskType: String, timeuuid: UUID) = Action.async { implicit request =>
+    val task = dao.getTask(year, TaskType.withName(taskType), timeuuid)
```

Comment:
But this is ok, it throws Exception just one time. It will mean that we have a bug. Programmer should NOT catch all possible exceptions everywhere, right? Exception here means that client code passes some thrash instead of legal taskType value. That means you or I made bug here some time. Then, someone has to fix this and immediately cover the test.